test/onhost-e2e/ansible/proxy.yaml

```diff
@@ -0,0 +1,171 @@
+---
+- name: Install Agent Control on a network restricted environment
+  hosts: 127.0.0.1
+  connection: local
+  become: true
+  gather_facts: yes
+
+  tasks:
+    - name: Cleanup
+      include_tasks: ./tasks/clean_all.yaml 
+
+    - name: Install docker
+      include_role:
+        name: caos.ansible_roles.docker_install 
+    
+    - name: Run proxy container
+      community.docker.docker_container:
+        name: "mitmproxy"
+        image: "mitmproxy/mitmproxy:12"
+        auto_remove: true
+        detach: true
+        tty: true
+        command: mitmweb --web-host 0.0.0.0
+        volumes:
+          - /tmp/mitmproxy:/home/mitmproxy/.mitmproxy
+        ports:
+          - "8080:8080"
+          - "8081:8081"
+
+    - name: Wait for mitmproxy CA certificate to be generated
+      ansible.builtin.wait_for:
+        path: /tmp/mitmproxy/mitmproxy-ca.pem
+        timeout: 30
+
+    - name: Copy mitmproxy CA certificate to system trust store
+      ansible.builtin.copy:
+        src: /tmp/mitmproxy/mitmproxy-ca.pem
+        dest: /usr/local/share/ca-certificates/mitmproxy-ca.crt
+        remote_src: yes
+
+    - name: Update CA certificates
+      ansible.builtin.command:
+        cmd: update-ca-certificates
+
+    - name: Install UFW
+      ansible.builtin.package:
+        name: ufw
+        state: present
+
+    - name: Backup current UFW rules
+      shell: |
+        ufw status > /tmp/ufw_status.txt
+        cp /etc/ufw/user.rules /tmp/user.rules.backup
+        cp /etc/ufw/user6.rules /tmp/user6.rules.backup
+
+    - name: Set firewall rules
+      block:
+        - name: Deny all outgoing traffic by default
+          community.general.ufw:
+            policy: deny
+            direction: outgoing
+
+        - name: Allow proxy to reach out from docker
+          community.general.ufw:
+            rule: allow
+            direction: out
+            interface: docker0
+
+        - name: Enable UFW
+          community.general.ufw:
+            state: enabled
+
+        - name: Verify UFW rules with bash script
+          shell: |
+            # Test that direct traffic is blocked
+            curl -s --max-time 5 https://www.newrelic.com
+            direct=$?
+            if [ $direct -eq 0 ]; then
+              echo "ERROR: Direct traffic is not blocked"
+              exit 1
+            fi
+            
+            # Test that proxied traffic works
+            curl -s --max-time 5 --proxy http://localhost:8080 https://www.newrelic.com
+            proxy=$?
+            if [ $proxy -ne 0 ]; then
+              echo "ERROR: Traffic blocked with proxy or proxy not working"
+              exit 1
+            fi
+
+        - name: Install Agent Control
```

Comment:
I see this `Install Agent Control` is a step inside `Set firewall rules`. Should it be outside? They are two different things.

Reply:
well the idea is to catch issues on the installation recipe to. For instance if tomorrow we add some other cli in there as part of the installation and we miss the proxy this test will fail. We have some good experience about this with the k8s one :)